Feature/core 426 switch profile type

Tschuck opened the pull request.

Commit:
add new card usage; add profile type switch modal

- [CORE-426]

Commit:
Merge branch 'FEATURE-498-form-data-component' into feature/CORE-426-switch-profile-type

Commit:
add registration form and company switch modal

- [CORE-426]

Commit:
merge

Commit:
add company forms; add device forms

- [CORE-426]

Commit:
add dispatcher and first profile migration lib

- [CORE-426]

Commit:
Merge branch 'feature/CORE-426-switch-profile-type' of github.com:evannetwork/ui-core-dapps into feature/CORE-426-switch-profile-type

Commit:
add save, update, load from profile for company account

- [CORE-426]

Commit:
fix profile.vue main page and verification responsive style

- [CORE-426]

Commit:
Merge branch 'feature/CORE-426-switch-profile-type' of github.com:evannetwork/ui-core-dapps into feature/CORE-426-switch-profile-type

Commit:
fix false used function

- [CORE-426]

Commit:
add dispatcher for device details update

Commit:
add dispatcher watchers for new profile (registration, type, contact)

- [CORE-426]

S3bb1 commented on dapps/profile.vue/src/i18n/de.ts:
```diff
@@ -36,6 +36,62 @@ const i18n: any = {
+    "company": {
+      "contact": {
+        "city": {
+          "error": "Bitte geben Sie den Städtenamen ein!",
```
```suggestion
          "error": "Bitte geben Sie einen Städtenamen ein!",
```

karladler replied:
```suggestion
          "error": "Bitte geben Sie einen Ort ein!",
```

S3bb1 commented on dapps/profile.vue/src/i18n/de.ts:
```diff
@@ -36,6 +36,62 @@ const i18n: any = {
+          "placeholder": "Geben Sie Ihre Postleitzahl ein"
+        },
+        "streetAndNumber": {
+          "error": "Bitte geben Sie eine Straße und eine Nummer ein!",
```
```suggestion
          "error": "Bitte geben Sie eine Straße und eine Hausnummer ein!",
```

S3bb1 commented on dapps/profile.vue/src/i18n/de.ts:
```diff
@@ -36,6 +36,62 @@ const i18n: any = {
+        "streetAndNumber": {
+          "error": "Bitte geben Sie eine Straße und eine Nummer ein!",
+          "label": "Straße und Hausnummer *",
+          "placeholder": "Geben Sie Ihre Straße und Nummer ein"
```
```suggestion
          "placeholder": "Geben Sie Ihre Straße und Hausnummer ein"
```

S3bb1 commented on dapps/profile.vue/src/i18n/de.ts:
```diff
@@ -36,6 +36,62 @@ const i18n: any = {
+        },
+        "register": {
+          "error": "Bitte wählen Sie den Registertyp aus!",
+          "label": "Registrieren *",
```
```suggestion
          "label": "Register *",
```

S3bb1 commented on dapps/profile.vue/src/i18n/de.ts:
```diff
@@ -36,6 +36,62 @@ const i18n: any = {
+        "registerNumber": {
+          "error": "Bitte geben Sie die Registernummer ein.",
+          "label": "Registernummer *",
+          "placeholder": "Wählen Sie..."
```
```suggestion
          "placeholder": "Registernummer des Unternehmens"
```

S3bb1 commented on dapps/profile.vue/src/i18n/de.ts:
```diff
@@ -44,7 +100,49 @@ const i18n: any = {
+    "device": {
+      "detail": {
+        "dataStreamSettings": {
+          "error": "Bitte fügen Sie Datenstrom-Einstellungsdateien ein!",
```
```suggestion
          "error": "Bitte fügen Sie IoT-Einstellungsdateien ein!",
```

karladler replied:
I asked myself WTH are "Datenstrom-Einstellungsdaten" :D

S3bb1 commented on dapps/profile.vue/src/i18n/de.ts:
```diff
@@ -44,7 +100,49 @@ const i18n: any = {
+      "detail": {
+        "dataStreamSettings": {
+          "error": "Bitte fügen Sie Datenstrom-Einstellungsdateien ein!",
+          "label": "Datenstromeinstellungen *",
```
```suggestion
          "label": "IoT Einstellungen *",
```

S3bb1 commented on dapps/profile.vue/src/i18n/de.ts:
```diff
@@ -44,7 +100,49 @@ const i18n: any = {
+        "dataStreamSettings": {
+          "error": "Bitte fügen Sie Datenstrom-Einstellungsdateien ein!",
+          "label": "Datenstromeinstellungen *",
+          "placeholder": "Hinzufügen von Datenstrom-Dateien"
```
```suggestion
          "placeholder": "Hinzufügen von IoT Einstellungs-Dateien"
```

S3bb1 commented on dapps/profile.vue/src/i18n/de.ts:
```diff
@@ -44,7 +100,49 @@ const i18n: any = {
+        "location": {
+          "error": "Bitte geben Sie einen Standort an!",
+          "label": "Ort *",
+          "placeholder": "Geben Sie die Geräteplatzkoordinaten ein."
```
```suggestion
          "placeholder": "Geben Sie die Standort Koordinaten ein."
```

karladler replied:
if we request coordinates we should offer at least in FE two input fields

S3bb1 commented on dapps/profile.vue/src/i18n/de.ts:
```diff
@@ -44,7 +100,49 @@ const i18n: any = {
+        },
+        "location": {
+          "error": "Bitte geben Sie einen Standort an!",
+          "label": "Ort *",
```
```suggestion
          "label": "Standort *",
```

S3bb1 commented on dapps/profile.vue/src/i18n/en.ts:
```diff
@@ -36,6 +36,62 @@ const i18n: any = {
+          "placeholder": "Choose..."
+        },
+        "salesTaxID": {
+          "error": "Please enter the Sales tax ID",
```
```suggestion
          "error": "Please enter the sales tax ID",
```

S3bb1 reviewed (changes requested):
typos :)

karladler commented on dapps/profile.vue/src/components/profile/company/registration/registration.ts:
```diff
@@ -0,0 +1,145 @@
+          type: 'select',
+          attr: {
+            options: [
+              { value: 'germany', label: '_profile.company.registration.countries.germany', }
```
maybe we can put country names in a more general app wide scope?

karladler replied:
why is actually only Germany allowed for now?

karladler commented on dapps/profile.vue/src/components/profile/company/registration/registration.ts:
```diff
@@ -0,0 +1,145 @@
+
+  async changeProfileData() {
+    dispatchers.updateProfileDispatcher.start((<any>this).getRuntime(), {
+      formData: this.registrationForm.toObject(),
```
what's doeas `toObject()` ?

Tschuck replied:
It's a function that maps all controls with it's names as keys to the current control values as values into one object.

karladler commented on dapps/profile.vue/src/components/profile/company/registration/registration.vue:
```diff
@@ -0,0 +1,43 @@
+</template>
+
+<script lang="ts">
+  import Component from './registration.ts';
```
`.ts`throws linter error

karladler commented on dapps/profile.vue/src/components/profile/device/detail/detail.ts:
```diff
@@ -0,0 +1,186 @@
+   */
+  async loadProfileData() {
+    const runtime = (<any>this).getRuntime();
+    const profileContract = runtime.profile.profileContract;
```
linter would prefer destructuring syntax: `const { profileContract } = runtime.profile;`

karladler commented on dapps/profile.vue/src/components/profile/device/detail/detail.ts:
```diff
@@ -0,0 +1,186 @@
+  /**
+   * Load the profile data an specify the registration form.
+   */
+  async loadProfileData() {
```
is this function special for devices or can we use it also for other profile types?

karladler commented on dapps/profile.vue/src/components/profile/device/detail/detail.vue:
```diff
@@ -0,0 +1,43 @@
+</template>
+
+<script lang="ts">
+  import Component from './detail.ts';
```
`.ts` ;)

karladler commented on dapps/profile.vue/src/components/profile/profile.ts:
```diff
@@ -90,4 +127,14 @@ export default class ProfileDetailComponent extends mixins(EvanComponent) {
+   */
+  typeSwitchModal() {
+    if (this.type === 'unspecified' &&
+        !(this as any).dispatcher.curr.running.updateProfileDispatcher) {
```
maybe put this in an `isLoading` component state?

karladler commented on dapps/profile.vue/src/components/profile/profile.ts:
```diff
@@ -90,4 +127,14 @@ export default class ProfileDetailComponent extends mixins(EvanComponent) {
+  typeSwitchModal() {
+    if (this.type === 'unspecified' &&
+        !(this as any).dispatcher.curr.running.updateProfileDispatcher) {
+      (this as any).$refs.profileType.$refs.modal.show();
```
maybe pass `$refs.modal` as property to the component?

karladler commented on dapps/profile.vue/src/components/profile/profile.vue:
```diff
@@ -55,21 +60,38 @@
-          
+        <div class="col-xl-8 mt-3">
+          <div class="text-center" v-if="type === 'unspecified'">
+            <template v-if="dispatcher.curr.running.updateProfileDispatcher">
```
v-if="isLoading" ?

karladler commented on dapps/profile.vue/src/components/profile/type/type.ts:
```diff
@@ -0,0 +1,80 @@
+// vue imports
+import Vue from 'vue';
+import Component, { mixins } from 'vue-class-component';
+import { Prop } from 'vue-property-decorator';
```
not needed?

karladler commented on dapps/profile.vue/src/components/profile/type/type.ts:
```diff
@@ -0,0 +1,80 @@
+  /**
+   * All selectable types
+   */
+  types = [
```
what about user?

karladler commented on dapps/profile.vue/src/components/profile/type/type.vue:
```diff
@@ -0,0 +1,70 @@
+              'ml-3': index !== 0,
+              'evan-highlight active': profileType === type,
+            }"
+            :highlight="false"
```
should be either not bound or not `false`

karladler commented on dapps/profile.vue/src/components/profile/type/type.vue:
```diff
@@ -0,0 +1,70 @@
+          <button type="button" class="btn btn-primary "
+            @click="typeChanged();">
+            {{ `_profile.type.change` | translate }}
+            <div class="spinner-border spinner-border-sm text-light ml-3" v-if="issuing"></div>
```
`issuing` seems not to be defined in type.ts

karladler commented on dapps/profile.vue/src/components/verifications/notary/notary.vue:
```diff
@@ -44,6 +44,22 @@
+            {{ '_profile.verifications.notary.request.request-ident' | translate }}
+          </evan-button>
+          <evan-button type="link" size="sm"
+            class="d-block  mt-1 text-muted"
```
two spaces

karladler commented on dapps/profile.vue/src/components/verifications/notary/topic-display/topic-display.vue:
```diff
@@ -29,9 +29,7 @@
-      <svg width="60" height="56" viewBox="0 0 60 56" fill="none">
-        <path d="M59.3333 28L52.8267 20.5867L53.7333 10.7733L44.1067 8.58666L39.0667 0.106659L30 3.99999L20.9333 0.106659L15.8933 8.58666L6.26666 10.7467L7.17333 20.56L0.666664 28L7.17333 35.4133L6.26666 45.2533L15.8933 47.44L20.9333 55.92L30 52L39.0667 55.8933L44.1067 47.4133L53.7333 45.2267L52.8267 35.4133L59.3333 28ZM24.6667 41.3333L14 30.6667L17.76 26.9067L24.6667 33.7867L48.92 9.67999L53.733 ...
-      </svg>
+      <i class="mdi mdi-check-decagram"></i>
```
shall we use the new verification icon from the assets here and everywhere else instead of the mdi-icon?

karladler commented on dapps/profile.vue/src/dispatchers/profile/updateProfile.ts:
```diff
@@ -0,0 +1,102 @@
+    );
+
+    const fileFields = ['settings', 'type'];
+
```
a lot of empty lines in this file :)

karladler commented on dapps/profile.vue/src/dispatchers/profile/updateProfile.ts:
```diff
@@ -0,0 +1,102 @@
+    await Promise.all(fileFields.map(async (field) => {
+      const blobs = []
+      if (data.formData[field]) {
+        for (let i = 0; i < data.formData[field].length; i++) {
```
maybe use es6 for of loop?

karladler commented on dapps/profile.vue/src/i18n/de.ts:
```diff
@@ -36,6 +36,62 @@ const i18n: any = {
+      "contact": {
+        "city": {
+          "error": "Bitte geben Sie den Städtenamen ein!",
+          "label": "Stadt *",
```
would use `Ort` in general instead of Stadt.

Tschuck replied:
```suggestion
          "label": "Ort *",
```

karladler commented on dapps/profile.vue/src/i18n/de.ts:
```diff
@@ -36,6 +36,62 @@ const i18n: any = {
+        },
+        "homepage": {
+          "error": "Bitte geben Sie eine gültige URL ein!",
+          "label": "Homepage *",
```
Website or Webseite seems more professional I would say. ;)

karladler commented on dapps/profile.vue/src/i18n/de.ts:
```diff
@@ -36,6 +36,62 @@ const i18n: any = {
+        "company": {
+          "error": "Bitte geben Sie einen Firmennamen ein!",
+          "label": "Unternehmen *",
+          "placeholder": "Geben Sie Ihren Firmennamen ein....."
```
a lot of dots ;)

karladler commented on dapps/profile.vue/src/i18n/de.ts:
```diff
@@ -44,7 +100,49 @@ const i18n: any = {
+          "placeholder": "Geben Sie den Gerätehersteller ein"
+        },
+        "owner": {
+          "error": "Bitte geben Sie eine korrekte Account Adresse an!",
```
Account-Adresse or Accountadresse

Commit:
fix card sizing in overview

- [CORE-426]

Commit:
add changes from review

- remove required markers from profile type formulars
- fix type changing
- make file upload for iot device optional
- allow to leave captcha code empty for test mode
- [CORE-426]

Commit:
remove ts file extension from vue imports

- [CORE-426]

karladler commented on dapps/profile.vue/src/components/profile/company/contact/contact.ts:
```diff
@@ -0,0 +1,130 @@
+      postalCode: {
+        value: contactData.postalCode || '',
+        validate: function(vueInstance: CompanyRegistrationForm, form: ContactFormInterface) {
+          return !!this.value.match(/^\d{5}$/);
```
u can use `/^\d{5}$/.test(this.value`) it returns the boolean directly. + works also when value is undefined ;)

karladler commented on dapps/profile.vue/src/components/profile/company/contact/contact.ts:
```diff
@@ -0,0 +1,130 @@
+      homepage: {
+        value: contactData.homepage || '',
+        validate: function(vueInstance: CompanyRegistrationForm, form: ContactFormInterface) {
+          return !!this.value.match(/((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=\+\$,\w]+@)?[A-Za-z0-9.-]+|(?:www.|[-;:&=\+\$,\w]+@)[A-Za-z0-9.-]+)((?:\/[\+~%\/.\w-_]*)?\??(?:[-\+=&;%@.\w_]*)#?(?:[\w]*))?)/);
```
thats a regex! is it intentional, that it would also matches e-mail addresses?

ps: here also the .test() syntax

Tschuck replied:
Ahh thats a pitty. I added another one ;)

Commit:
fix pr review comments

- load countries from external lib
- [CORE-426]

Commit:
Merge branch 'develop' into feature/CORE-426-switch-profile-type

Commit:
add pr review fixes

- i18n typos
- use correct regex for urls and postcodes
- refactor form.toObject
- [CORE-426]

Commit:
Merge branch 'feature/CORE-426-switch-profile-type' of github.com:evannetwork/ui-core-dapps into feature/CORE-426-switch-profile-type
